Automattic/jetpack
# Jetpack AI Image: create first draft of the general purpose generator

Fixes #37781.

## Proposed changes:

* Rename main directory from `featured-image` to `ai-image` so it is generic
* Reorganize file names to make room for more than one type of generation
* Create `GeneralPurposeImage` generator as a copy of the `FeaturedImage` generator, only with different titles
* Add entrypoint for the general purpose generation on the external media dropdown
   * refer to the feature beta flag to enable/disable it - WIP

The objective of this PR is to create a basic structure where the new generator is already in place, but hidden behind a beta flag, so we can iterate on it and keep merging/improving things as we go, so no big merge/change is necessary when things are ready (details: pe4Cmx-2pm-p2).

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






First let's ensure the Featured Image generator is working as before:

- add some content to the block editor and use the featured image generation tool from:
   - the AI Assistant section on the sidebar (the button that says "Generate image")
   - the featured image section (that may be on the Summary panel or on it's own pane, both on the Settings sidebar
- confirm the tool is working as before:
   - you can generate the first image
   - you can ask for new images, both with and without a custom prompt
   - you can navigate between the images
   - you can set some image as featured

Now let's confirm the new GeneralPurposeImage generator component is in place. I added a basic entrypoint on the image block, so let's use it to test.

- make sure your site has beta extensions enabled
- add some content to the block editor and then add an image block below it
- on the image block, click "Select image" and then click "Generate with AI" on the dropdown menu that will show up:

<img width="600" alt="Screenshot 2024-06-11 at 11 51 31" src="https://github.com/Automattic/jetpack/assets/6760046/30b2217b-6a86-4352-a9e8-f01237fe6644">

- when you click it, the generator modal will show up already generating a featured image (since the code is identical to the featured image generator)
- confirm the tool show the "Generate image with AI" title (instead of "Generate featured image with AI"), this will prove we are using the new component:

<img width="600" alt="Screenshot 2024-06-11 at 11 52 00" src="https://github.com/Automattic/jetpack/assets/6760046/ad112d1b-95b2-4ea5-8ffd-2c3ca055abdd">

- confirm the tool works in the same way the featured image generator works:
   - you can generate the first image
   - you can ask for new images, both with and without a custom prompt
   - you can navigate between the images
   - you can set some image as featured

Finally, let's confirm the beta flag is working as expected:

- disable the beta extensions (set it to production) and then go back to the editor
- confirm the option to "Generate with AI" is NOT present on the image block, so it's hidden for production users:

<img width="600" alt="Screenshot 2024-06-11 at 12 13 06" src="https://github.com/Automattic/jetpack/assets/6760046/e189d35f-e916-4f94-8e3f-7ff1dfd3b9ba">
